pkg/virtctl/imageupload/imageupload.go

```diff
@@ -276,7 +276,22 @@ func waitUploadPodRunning(client kubernetes.Interface, namespace, name string, i
 
 		podPhase, _ := pvc.Annotations[PodPhaseAnnotation]
 
-		done := (podPhase == string(v1.PodRunning)) && (len(endpoints.Subsets) > 0)
+		done := false
+		availableEndpoint := false
+		for _, subset := range endpoints.Subsets {
+			if len(subset.Addresses) > 0 {
```

Comment:
Ah! So the major change here is to not only check for endpoints, but to verify the endpoint is addressable.